src/main/java/com/microsoft/sqlserver/jdbc/KerbAuthentication.java

```diff
@@ -390,6 +391,7 @@ private String findRealmFromHostname(RealmValidator realmValidator,
             int port,
             GSSCredential ImpersonatedUserCred) throws SQLServerException {
         this(con, address, port);
+        isUserCreatedCredential = true;
```

Comment:
Can we pass this flag to the constructor from `logon()`? 
` if (null != ImpersonatedUserCred)`

Reply:
changes pushed